protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -2488,6 +2504,21 @@ func getMessageOpenAPIOption(reg *descriptor.Registry, msg *descriptor.Message)
 	return opts, nil
 }
 
+func getServiceOpenAPIOption(reg *descriptor.Registry, svc *descriptor.Service) (*openapi_options.Tag, error) {
+	opts, err := extractTagOptionFromServiceDescriptor(svc.ServiceDescriptorProto)
+	if err != nil {
+		return nil, err
+	}
+	if opts != nil {
+		return opts, nil
+	}
```

Comment:
This gives the protobuf file annotations precedence over the openapi configuration file, is that what we want? I think users usually use the openapi configuration file when they can't edit the protobuf file for some reason, so maybe the precedence should be in the other direction, so that an openapi configuration file can overwrite what's set in the protobuf file?

Reply:
Ah Right. I'll fix it so that the options defined in the openapi configuration file take precedence.

Reply:
I found that the options defined in the **proto** file always overwrite openapi configurations for the part of Method, Message, File and Field.

Example (for Method):
https://github.com/grpc-ecosystem/grpc-gateway/blob/c12472a456fecdd504ccbb8eb645776c3c05f073/protoc-gen-openapiv2/internal/genopenapi/template.go#L2465

I think we need to unify which configuration file has priority, but I feel that giving priority to openapi would have a not small impact on the existing output for Method, Message, File and Field.

How about creating a new issue for the fix to change the priority of the Proto and OpenAPI configuration files, and scope this fix to support the Service tag defined in the OpenAPI configuration file?

Reply:
If it is better to change the priority of the Proto and OpenAPI configuration files only for `Service` in advance, I'll fix it as planned.

Reply:
Sounds good!

Reply:
Thank you for your confirmation!

Just to be sure, do you want me to change the priority of configs only for Service in advance? Or do  we not change it in this PR, but handle it in conjunction with other changes?

Reply:
Handle the service here, and lets do the others in a separate PR.

Reply:
Thanks. I've fixed the priority of config files only for Service.